src/main/java/uk/gov/hmcts/reform/slc/services/ReportParser.java

```diff
@@ -0,0 +1,67 @@
+package uk.gov.hmcts.reform.slc.services;
+
+import org.apache.commons.csv.CSVFormat;
+import org.apache.commons.csv.CSVParser;
+import org.apache.commons.csv.CSVRecord;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.slc.model.LetterPrintStatus;
+import uk.gov.hmcts.reform.slc.services.steps.getpdf.FileNameHelper;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.time.ZonedDateTime;
+import java.util.List;
+
+import static java.util.stream.Collectors.toList;
+import static java.util.stream.StreamSupport.stream;
+
+@Component
+public class ReportParser {
+
+    private static final Logger logger = LoggerFactory.getLogger(ReportParser.class);
+
+    public List<LetterPrintStatus> parse(byte[] report) {
+        try (CSVParser parser = parserFor(report)) {
+
+            return stream(parser.spliterator(), false)
+                .map(row -> toPrintStatus(row))
```

Comment:
we can just use signature over here as `this::toPrintStatus`

Reply:
using lambda on purpose so that it's clearer what's being passed

Reply:
I also think it's cleaner as it is